components/wmtk_components/mesh_decimation/wmtk/components/mesh_decimation/internal/MeshDecimation.cpp

```diff
@@ -0,0 +1,172 @@
+            acc_edge.scalar_attribute(edge) = 1;
+        }
+        acc_len.scalar_attribute(edge) = (acc_pos.vector_attribute(edge) -
+                                          acc_pos.vector_attribute(m_mesh.switch_tuple(edge, PV)))
+                                             .norm();
+    }
+
+    auto op_scaffold = std::make_shared<operations::EdgeCollapse>(m_mesh);
+
+    op_scaffold->add_invariant(
+        std::make_shared<TodoInvariant>(m_mesh, edge_handle.as<int64_t>(), 0));
+    op_scaffold->add_invariant(std::make_shared<TodoSmallerInvariant>(
+        m_mesh,
+        edge_len_handle.as<double>(),
+        4.0 / 5.0 * m_target_len));
+
+    auto m_amips = std::make_shared<function::AMIPS>(m_mesh, position);
+    auto m_link_conditions = std::make_shared<InvariantCollection>(m_mesh);
+    m_link_conditions->add(std::make_shared<MultiMeshLinkConditionInvariant>(m_mesh));
+    auto m_function_invariant =
+        std::make_shared<SmallerFunctionInvariant>(m_mesh.top_simplex_type(), m_amips, 30);
+    auto m_inversion_invariant =
+        std::make_shared<SimplexInversionInvariant>(m_mesh, position.as<double>());
+
+    // Edge length update
+    auto compute_edge_length = [](const Eigen::MatrixXd& P) -> Eigen::VectorXd {
+        assert(P.cols() == 2);
+        assert(P.rows() == 2 || P.rows() == 3);
+        return Eigen::VectorXd::Constant(1, (P.col(0) - P.col(1)).norm());
+    };
+    auto m_edge_length_update =
+        std::make_shared<wmtk::operations::SingleAttributeTransferStrategy<double, double>>(
+            edge_len_handle,
+            position,
+            compute_edge_length);
+    auto m_prio_short_edges_first = [&](const simplex::Simplex& s) {
+        assert(s.primitive_type() == PrimitiveType::Edge);
+        auto acc = m_mesh.create_accessor<double>(edge_len_handle);
+        return std::vector<double>({acc.scalar_attribute(s.tuple())});
+    };
+
+
+    op_scaffold->add_invariant(m_link_conditions);
+    op_scaffold->add_invariant(m_function_invariant);
+    op_scaffold->add_invariant(m_inversion_invariant);
+
+    op_scaffold->add_transfer_strategy(m_edge_length_update);
+
+    op_scaffold->set_priority(m_prio_short_edges_first);
+
+    op_scaffold->set_new_attribute_strategy(
+        position,
+        wmtk::operations::CollapseBasicStrategy::Mean);
+    op_scaffold->set_new_attribute_strategy(vertex_handle);
+
+    op_scaffold->set_new_attribute_strategy(edge_handle);
+    op_scaffold->set_new_attribute_strategy(edge_len_handle);
+    op_scaffold->set_new_attribute_strategy(cell_tag_handle);
+
+    // pass_through
+    for (const auto& attr : m_pass_through_attributes) {
+        op_scaffold->set_new_attribute_strategy(attr);
+    }
+
+    while (true) {
+        Scheduler scheduler;
+        SchedulerStats pass_stats = scheduler.run_operation_on_all(*op_scaffold);
+        if (pass_stats.number_of_successful_operations() == 0) break;
```

Comment:
Please do not write one liner `if`s. Always add brackets around the code inside the if: `{ break; }`.